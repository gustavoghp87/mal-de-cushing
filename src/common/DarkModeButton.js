import styles from '../css/Home.module.css'
import { PropTypes } from 'prop-types'

export const DarkModeButton = ({ isEnglish, isMobile, isDarkMode, setIsDarkModeHandler }) => {
    
    return (
        <div
            className={`${styles.card} custom-control custom-switch ${isDarkMode ? 'bg-dark' : 'bg-light'} `}
            style={{
                position: 'fixed',
                bottom: isMobile ? '' : 0,
                top: isMobile ? 0 : '',
                left: isMobile ? '40%' : 0,
                marginLeft: '3%',
                cursor: 'pointer',
                padding: '10px'
            }}
        >
            &nbsp;
            &nbsp;
            &nbsp;
            &nbsp;
            &nbsp;
            
            <input className={'custom-control-input'}
                type={'checkbox'}
                id={"customSwitches"}
                checked={isDarkMode}
                onChange={() => {
                    window.localStorage.setItem('darkModeCushing', (!isDarkMode).toString())
                    setIsDarkModeHandler(!isDarkMode)
                }}
            />
            
            <label
                className={`custom-control-label ${isDarkMode ? 'text-white' : ''}`}
                htmlFor={'customSwitches'}
                style={{
                    cursor: 'pointer'
                }}
            >
                <b> {isEnglish && isDarkMode && (isMobile ? 'Mode' : 'Light Mode')} </b>
                <b> {isEnglish && !isDarkMode && (isMobile ? 'Mode' : 'Dark Mode')} </b>
                <b> {!isEnglish && isDarkMode && (isMobile ? 'Modo' : 'Modo Claro')} </b>
                <b> {!isEnglish && !isDarkMode && (isMobile ? 'Modo' : 'Modo Oscuro')} </b>
            </label>
            
        </div>
    )
}

DarkModeButton.propTypes = {
    isEnglish: PropTypes.bool.isRequired,
    isMobile: PropTypes.bool.isRequired,
    isDarkMode: PropTypes.bool.isRequired,
    setIsDarkModeHandler: PropTypes.func.isRequired
}

DarkModeButton.defaultProps = {
    isDarkMode: true
}
