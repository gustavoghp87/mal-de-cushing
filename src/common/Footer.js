import styles from '../css/Home.module.css'

export const Footer = () => (
    <footer className={styles.footer}>
        <p> Se aceptan donaciones en Bitcoin para pagar el dominio: </p>

        <p> 1AtHe23t5zHWPS5wSYtDXz2fFjZqB9FgHN </p>

        <a
            href={"https://www.maldecushing.com"}
            target={'_blank'}
            rel={'noopener noreferrer'}
        >
            maldecushing.com
            <span className={styles.logo}>
                {/* <Image src="/vercel.svg" alt="Logo" width={72} height={16} /> */}
            </span>
        </a>
    </footer>
)