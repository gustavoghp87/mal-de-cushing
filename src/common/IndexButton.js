import styles from '../css/Home.module.css'
import { PropTypes } from 'prop-types'

export const IndexButton = ({ isEnglish, isMobile, isDarkMode, setShowIndexHandler }) => {

    return (
    <>
        <div
            className={`${styles.card} ${isDarkMode ? 'bg-dark text-white' : 'bg-light'}`}
            style={{
                position: 'fixed',
                left: 0,
                top: 0,
                marginLeft: '3%',
                cursor: 'pointer',
                padding: isMobile ? '10px 37px' : ''
            }}
            onClick={() => setShowIndexHandler(true)}
        >

            <b className={isMobile ? 'd-none' : ''}> {isEnglish ? "Open Index" : "Ver Índice"} </b>
            <b className={isMobile ? '' : 'd-none'}> {isEnglish ? "Index" : "Índice"} </b>
            
        </div>

    </>
    )
}

IndexButton.propTypes = {
    isEnglish: PropTypes.bool.isRequired,
    isMobile: PropTypes.bool.isRequired,
    isDarkMode: PropTypes.bool.isRequired,
    setShowIndexHandler: PropTypes.func.isRequired
}

IndexButton.defaultProps = {
    isDarkMode: true
}
