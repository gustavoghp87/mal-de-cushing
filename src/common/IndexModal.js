import { Modal, Button } from 'react-bootstrap'
import { PropTypes } from 'prop-types'

export const IndexModal = ({ isEnglish, isMobile, isDarkMode, showIndex, setShowIndexHandler }) => {
    
    const IndexComponent = () => {

        const sections = ['1','2','3','4','5','6','7','8','9','10','11','12','13','14','15','16','17','18','19','20','21','22','23','24','25', '26']
    
        return(
        <>
            {sections.map((section, index) => (
                <div key={index}>
                    <a href={"#" + section} type={'button'}>
                        <h6> {document?.getElementById(section)?.innerText} </h6>
                    </a>
                </div>
            ))}
        </>
        )
    }

    return (
        <Modal
            show={showIndex}
            onHide={() => setShowIndexHandler(false)}
            
        >

            <Modal.Header closeButton className={isDarkMode ? 'bg-dark text-white' : ''}>
                <Modal.Title> {isEnglish ? "Index" : "√çndice"} </Modal.Title>
            </Modal.Header>

            <Modal.Body
                className={isDarkMode ? 'bg-dark text-white' : ''}
                onClick={() => setShowIndexHandler(false)}
            >
                <IndexComponent />
            </Modal.Body>

            <Modal.Footer className={isDarkMode ? 'bg-dark text-white' : ''}>
                <Button variant={'danger'} onClick={() => setShowIndexHandler(false)}>
                    {isEnglish ? "Close" : "Cerrar"}
                </Button>
            </Modal.Footer>
        </Modal>
    )
}

IndexModal.propTypes = {
    isEnglish: PropTypes.bool.isRequired,
    isMobile: PropTypes.bool.isRequired,
    isDarkMode: PropTypes.bool.isRequired,
    showIndex: PropTypes.bool.isRequired,
    setShowIndexHandler: PropTypes.func.isRequired
}

IndexModal.defaultProps = {
    isDarkMode: true
}