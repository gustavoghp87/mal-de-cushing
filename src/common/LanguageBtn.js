import { PropTypes } from 'prop-types'
import { ArrowBarRight } from 'react-bootstrap-icons'
import styles from '../css/styles.module.css'

export const LanguageBtn = ({ isEnglish, setIsEnglish }) => {

    return (
        <div style={{ paddingTop: '90px' }}>
            <div className={styles.card + ' d-block m-auto'} onClick={() => setIsEnglish(!isEnglish)}>
                <h2 className={'mb-0 text-center'} style={{ width: '250px' }}>
                    {isEnglish ? "Versión en Español" : "English version"}
                    &nbsp; <ArrowBarRight />
                </h2>
            </div>
        </div>
    )
}

LanguageBtn.propTypes = {
    isEnglish: PropTypes.bool.isRequired,
    setIsEnglish: PropTypes.func.isRequired
}
