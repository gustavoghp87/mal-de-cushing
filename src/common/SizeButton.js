import styles from '../css/Home.module.css'
import { PlusSquare, DashSquare } from 'react-bootstrap-icons'

export const SizeButton = props => {

    const { isDarkMode, isEnglish, isMobile } = props
    const textSize = window.localStorage.getItem('textSizeCushing')
    
    return (
        <div
            className={`${styles.card} ${isDarkMode ? 'bg-dark text-white' : 'bg-light'}`}
            style={{
                position: 'fixed',
                bottom: isMobile ? '' : 0,
                top: isMobile ? 0 : '',
                right: 0,
                marginRight: isMobile ? '3%' : '3%',
                padding: '5px'
            }}
        >

            <div className={isDarkMode ? 'text-white' : ''}>
                
                <div
                    className={`p-1 pl-2 ${textSize === '1.8rem' ? 'd-none' : ''}`}
                    onClick={() => {
                        if (textSize === '1.5rem') window.localStorage.setItem('textSizeCushing', "1.8rem")
                        else window.localStorage.setItem('textSizeCushing', "1.5rem")
                        window.location.reload()
                    }}
                    style={{ cursor: 'pointer' }}
                >
                    <PlusSquare size={26} />
                    &nbsp;
                    <b className={isMobile ? 'd-none' : ''}> {isEnglish ? 'Increase Text Size' : 'Agrandar Texto'} &nbsp; </b>
                    <b className={isMobile ? '' : 'd-none'}> A &nbsp; </b>
                    {/* color={'royalblue'} */}
                </div>

                <div
                    className={`p-1 pl-2 ${textSize === '1rem' ? 'd-none' : ''}`}
                    onClick={() => {
                        if (textSize === '1.8rem') window.localStorage.setItem('textSizeCushing', "1.5rem")
                        else window.localStorage.setItem('textSizeCushing', "1rem")
                        window.location.reload()
                    }}
                    style={{ cursor: 'pointer' }}
                >
                    <DashSquare size={26} />
                    &nbsp;
                    <b className={isMobile ? 'd-none' : ''}> {isEnglish ? 'Decrease Text Size' : 'Achicar Texto'} </b>
                    <b className={isMobile ? '' : 'd-none'}> A </b>
                </div>
            </div>

        </div>

    )
}
