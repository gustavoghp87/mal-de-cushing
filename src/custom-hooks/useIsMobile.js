import { useEffect, useState } from 'react'

export const useIsMobile = () => {
    const [isMobile, setIsMobile] = useState(window.innerWidth < 1000)

    const adjustIsMobile = () => {
        setTimeout(() => {
            const isMobile1 = window.innerWidth < 1000
            if (isMobile1 !== isMobile) setIsMobile(isMobile1)
            adjustIsMobile()
        }, 2000)
    }

    useEffect(() => {
        adjustIsMobile()
    }, [])


    return isMobile
}