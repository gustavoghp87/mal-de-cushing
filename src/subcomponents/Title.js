import { useSelector } from 'react-redux'

export const Title = () => {

    const { isEnglish } = useSelector(state => state.isEnglish)
    const { isMobile } = useSelector(state => state.isMobile)
    
    return (
        <div
            style={{
                margin: 0,
                lineHeight: 1.15,
                fontSize: isMobile ? '2.5rem ' : '3.4rem',
                textAlign: 'center',
                paddingTop: isMobile ? '80px' : '140px',
                marginBottom: '60px'
            }}
        >
            <p>
                {isEnglish ?
                    "Cushing's Desease in dogs and trilostane & mitotane treatments:"
                    :
                    "Mal de Cushing en perros y tratamientos con trilostano y mitotano:"
                }
            </p>

            <p>
                {isEnglish ?
                    "Ally's case"
                    :
                    "El caso de Ally"
                }
            </p>
        </div>
    )
}
