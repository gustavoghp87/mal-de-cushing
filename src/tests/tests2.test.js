import '@testing-library/jest-dom'
import { Title } from '../common/Title'
import { shallow } from 'enzyme'
import { useIsMobile } from '../custom-hooks/useIsMobile'
import { renderHook } from '@testing-library/react'
jest.mock('../custom-hooks/useIsMobile')

describe('tests 2', () => {

    let wrapper = shallow(<Title isEnglish={false} isMobile={false} />);
    
    beforeEach(() => {
        jest.clearAllMocks();
        console.log("Before each");
        wrapper = shallow(<Title isEnglish={false} isMobile={false} />);
    })

    test('Title component should have not errors when is rendered', () => {
        const expectedTitle = "Mal de Cushing en perros y tratamientos con trilostano y mitotano:El caso de Ally";
    
        const title = wrapper.find('div').text();

        const button = wrapper.find('button').at(0);
        try {
            button.simulate('click')
        } catch (error) {
            console.log("There's not any button");
        }

        const img = wrapper.find('img');
        const imgAttributes = img.props();
        const cssClasses = img.prop('className');

        const input = wrapper.find('input').at(0);
        input.simulate('change', { target: { value: 'value' } });

        const form = wrapper.find('form');
        expect(form.exists()).toBe(true);
        form.simulate('submit', { preventDefault: () => { } });

        const setIsMobile = jest.fn();
        expect(setIsMobile).not.toHaveBeenCalled();
        expect(setIsMobile).toHaveBeenCalledTimes(2);
        expect(setIsMobile).toHaveBeenCalledWith(expect.any(Function));


        console.log(imgAttributes, cssClasses);
    
        expect(wrapper).toMatchSnapshot();
        expect(title).toBe(expectedTitle)
    })

    test('Mocking', async () => {
        useIsMobile.mockReturnValue(false)
    })

    test('Hooks', async () => {
        const { result, waitForNextUpdate } = renderHook(() => useIsMobile());
        const { r } = result.current;
        console.log(r);
        await waitForNextUpdate();
        expect(r).toBe(false);
    })
})
