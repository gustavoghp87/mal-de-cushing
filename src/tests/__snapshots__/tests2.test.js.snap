// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests 2 Title component should have not errors when is rendered 1`] = `
<div
  style={
    Object {
      "fontSize": "3.4rem",
      "lineHeight": 1.15,
      "margin": 0,
      "textAlign": "center",
    }
  }
>
  Mal de Cushing en perros y tratamientos con trilostano y mitotano:
  <br />
  El caso de Ally
</div>
`;
